src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -16,383 +13,260 @@ public static class NUnitTestExtractorApp
                 {
-                    if (categoryNames != null && categoryNames.Contains(value))
-                    {
-                        return true;
-                    }
+                    continue;
                 }
-                return false;
-            }
-            throw new ArgumentNullException("excludeInfo", "no excludeInfo specified");
-        }
 
-        /// <summary>
-        /// Depending on number of args within the TestCase, function will return properly formatted testcase
-        /// </summary>
-        /// <param name="declaringType">declaring type of testcase method(namespace.class)</param>
-        /// <param name="name">name of testcase</param>
-        /// <param name="testCaseArgs">the object array containing the testcase's arguments</param>
-        /// <param name="data">the string which is appended to</param>
-        /// <returns>the properly formatted TestCase string</returns>
-        public static string FormattedTestCase(string declaringType, string name, object[] testCaseArgs, string data)
-        {
+                // match the "Include" category from the function category list
+                // if the function has matched include category, we will count this function as a valid test
+                if (string.IsNullOrWhiteSpace(_options.IncludeInfo)
+                    || categories.Any(x => x.ConstructorArguments.Any(y => y.Value.Equals(_options.IncludeInfo))))
+                {
+                    Console.WriteLine("Found the test and add it.");
 
-            if (testCaseArgs != null && testCaseArgs.Length > 0)
-            {
-                data = declaringType + "." + name + "(";
-                data = AppendArgumentsForTestCasesToString(testCaseArgs, data);
-                data += ")";
-            }
-            else if (testCaseArgs != null && testCaseArgs.Length == 0)
-            {
-                data = declaringType + "." + name + "()";
+                    tests.Add(test.DeclaringType.GetTypeInfo().Name + "." + test.Name);
+                }
             }
 
-            return data;
+            return tests;
         }
 
         /// <summary>
-        /// Parses string into proper Level
+        /// 
         /// </summary>
-        /// <param name="levelToParse">The string to parse</param>
-        /// <returns>The Level from parsing the string</returns>
-        public static Level ParseLevel(string levelToParse)
+        /// <param name="outputInfo"></param>
+        /// <param name="test"></param>
+        private static void modifyOutput(string outputInfo, MethodInfo test)
         {
-            Level level = Level.Null;
-            try
+            if (test == null)
             {
-                level = (Level)Enum.Parse(typeof(Level), levelToParse, ignoreCase: true);
+                throw new ArgumentNullException("test", "can't be NULL or Empty.");
```

Comment:
[Should] Use ThrowIf.ArgumentNull() instead of if & throw...  Ex.
`ThrowIf.ArgumentNull(test, nameof(test));`

Reply:
ok